src/content/docs/en/guides/testing.mdx

```diff
@@ -155,6 +155,96 @@ More information about Cypress can be found in the links below:
 - [Introduction to Cypress](https://docs.cypress.io/guides/core-concepts/introduction-to-cypress)
 - [Testing Your App](https://docs.cypress.io/guides/end-to-end-testing/testing-your-app)
 
+## NightwatchJS
+
+No-compromise test automation framework with a powerful set of tools to write, run and debug your tests across web, using Google Chrome, Firefox, Safari, and their mobile equivalents, as well as  native mobile applications.
+
+### Installation
+
+You can install NightwatchJS within your Astro project using the package manager of your choice. Follow the CLI steps to choose JavaScript/TypeScript, name your test folder, if you want to do component testing and if you want to test on mobile browsers.
+<PackageManagerTabs>
+  <Fragment slot="npm">
+  ```shell
+  npm init nightwatch@latest
+  ```
+  </Fragment>
+  <Fragment slot="pnpm">
+  ```shell
+  pnpm dlx create-nightwatch
+  ```
+  </Fragment>
+  <Fragment slot="yarn">
+  ```shell
+  yarn create nightwatch
+  ```
+  </Fragment>
+</PackageManagerTabs>
+
+### Create your first Nightwatch test
+
+1. Choose a page to test. This example will test the example page `index.astro` below.
+
+    ```html title="src/pages/index.astro"
+    ---
+    ---
+    <html lang="en">
+      <head>
+        <title>Astro is awesome!</title>
+        <meta name="description" content="Pull content from anywhere and serve it fast with Astro's next-gen island architecture." />
+      </head>
+      <body></body>
+    </html>
+    ```
+
+1. Create a test file in `test/`
+
+		```js
+		describe('Astro testing with Nightwatch', function () {
+		    before(browser => browser.navigateTo('http://localhost:4321/'));
+		
+		    it("check that the title is correct", function (browser) {
+		        browser.assert.titleEquals('Astro is awesome!')
+		    });
+		
+		    after(browser => browser.end());
+		});
+		```
+
+:::tip[Set a `baseUrl`]
+    You can set [`"baseURL": "http://localhost:4321"`](https://nightwatchjs.org/guide/reference/settings.html#setting-the-baseurl-property) in the `nightwatch.conf.js` configuration file to use `browser.navigateTo("/")` instead of `browser.navigateTo("http://localhost:4321/")` for a more convenient URL.
+:::
+
+### Running your NightwatchJS tests
+
+You can run a single test or several tests at once, testing one or multiple browsers. By default, your test results will be shown in the terminal. Optionally, you can open the HTML Test Reporter to show a full report and filter test results. You can run the tests with the [NightwatchJS VSCode Extension](https://marketplace.visualstudio.com/items?itemName=browserstackcom.nightwatch) or using the CLI steps below
+
+1. To run our test from the previous example using the command line use the follow. Optionally, include the file name to run just the single test:
```

Comment:
```suggestion
1. To run all tests, enter the following command in the terminal. Optionally, include the file name to run just the single test:
```
Is this correct? I know you have been following (very faithfully! thank you!) the existing Playwright example, but I think this makes more sense as a helpful instruction. (And, I'll likely go back and update that section myself later.)